test/distributed/_shard/sharded_tensor/ops/test_linear.py

```diff
@@ -190,7 +190,7 @@ def test_sharded_linear_errors(self):
         for spec in generate_chunk_sharding_specs_for_test(0):
             fc1 = torch.nn.Linear(10, 10).cuda(self.rank)
             shard_parameter(fc1, "bias", spec)
-            with self.assertRaisesRegex(TypeError, 'bias needs to be torch.Tensor'):
```

Comment:
This is not actually what we want to test here. Maybe instead of changing the error msg, can you add one more line of shard_parameter(fc1, "weight", spec) so that the error msg will be triggered?